### ISSUE TITLE
StringLiteralDuplication potential false positive with interpolated strings

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#7271]
## Expected Behavior
No finding.

## Observed Behavior
> Test.kt:76:21 (see cursor in screenshot): Multiple occurrences of the same string literal within a single file detected. Prefer extracting the string literal into a property or constant. [StringLiteralDuplication]

## Steps to Reproduce

![image](https://github.com/detekt/detekt/assets/2906988/2c25b66c-df1f-453d-861e-8a1aa8578609)

## Context

As you can see in the screenshot above, the dynamic test names are being detected as duplicate. This would be fine, because at first glance all the "cycle start-end" strings "look" the same, but if you look at the start/end variables, you can see they all have different types. At this point, I'm unsure if this detection is valid, because there would be no function that could be extracted to simplify the code.

## Your Environment

* Version of detekt used: 1.23.6
* Version of Gradle used (if applicable): 8.7
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: N/A
* Link to your project (if it's a public repository): N/A

**Comment by BraisGabin** [comment]
> At this point, I'm unsure if this detection is valid, because there would be no function that could be extracted to simplify the code.

A function that receives `Any`. You only care about the `toString` of those classes.

Anyway, should this rule check strings with variables inside it? I'm not sure.

**Comment by TWiStErRob** [comment]
🤦🏻‍♂️ that makes so much sense. So this is a viable option:
```kotlin
fun cycle(start: Any, end: Any): String =
    "cycle ${start}-${end}"
```

I guess your observation will hold true for any expression inside an interpolated string. So extracting a function with String or Any params is a viable way to fix the duplication.

Could this be just a doc update with a hint then? (TBH, I haven't read the docs of this rule yet, might be already there...)

**Comment by BraisGabin** [comment]
> Could this be just a doc update with a hint then?

No sure about what could we write there for this case but sure, any improvement on the docs is more than welcome.
